### PR TITLE
Asset parameterization workaround for EDC 0.2+

### PR DESCRIPTION
**Pull request by ununhexium**
https://github.com/sovity/edc-ui/issues/703
https://github.com/sovity/edc-ui/issues/707

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [x] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/TransferRequestBuilder.java [hunk 1]
```diff
@@ -38,14 +39,15 @@ public class TransferRequestBuilder {
...
+    private final ParameterizationCompatibilityUtils parameterizationCompatibilityUtils;
 
     public TransferRequest buildCustomTransferRequest(
             InitiateTransferRequest request
     ) {
         var contractId = request.getContractAgreementId();
         var agreement = contractAgreementUtils.findByIdOrThrow(contractId);
         var negotiation = contractNegotiationUtils.findByContractAgreementIdOrThrow(contractId);
-        var address = edcPropertyUtils.buildDataAddress(request.getDataSinkProperties());
+        var address = parameterizationCompatibilityUtils.enrich(edcPropertyUtils.buildDataAddress(request.getDataSinkProperties()), request.getTransferProcessProperties());
```
Comment (efiege) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/TransferRequestBuilder.java:50]: Also Test with the Custom Json Request

**Review by efiege — CHANGES_REQUESTED** [review]
Also write a test for the management API. Also using old paramter prefix and new prefixes.

File: tests/build.gradle.kts [hunk 2]
```diff
@@ -23,7 +25,11 @@ dependencies {
     testImplementation("net.javacrumbs.json-unit:json-unit-assertj:${jsonUnit}")
     testImplementation("org.mockito:mockito-core:${mockitoVersion}")
     testImplementation("org.assertj:assertj-core:${assertj}")
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.10.0")
+    testImplementation("org.junit.jupiter:junit-jupiter-api:${jupiterVersion}")
+    testImplementation("org.junit.jupiter:junit-jupiter-params:${jupiterVersion}")
+    testImplementation("org.mock-server:mockserver-netty:${httpMockServerVersion}") {
+        // TODO: increase minimum guava version
```
[the remark this answers is not in the document]
Comment (ununhexium) [reply tests/build.gradle.kts:31]: https://github.com/sovity/edc-extensions/issues/878

File: extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/ParameterizationCompatibilityUtils.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class ParameterizationCompatibilityUtils {
+    public DataAddress enrich(DataAddress dataAddress, Map<String, String> transferProcessProperties) {
+        if(transferProcessProperties == null) {
+            return dataAddress;
+        }
+
+        Map<String, String> mappings = Map.of(
```
Comment (efiege) [inline extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/ParameterizationCompatibilityUtils.java:30]: can be static in class

File: extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/ParameterizationCompatibilityUtils.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                "https://w3id.org/edc/v0.0.1/ns/mediaType", "https://sovity.de/workaround/proxy/param/mediaType",
+                "https://w3id.org/edc/v0.0.1/ns/contentType", "https://sovity.de/workaround/proxy/param/mediaType"
+        );
+
+        HashMap<String, Object> combined = new HashMap<>(dataAddress.getProperties());
+
+        for (val e : transferProcessProperties.entrySet()) {
+            val mapping = mappings.get(e.getKey());
+            if (mapping != null) {
+                combined.put(mapping, e.getValue());
```
Comment (efiege) [inline extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/ParameterizationCompatibilityUtils.java:44]:
```suggestion
                combined.put(workaroundProperty, transferProcessProperty.getValue());
```

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/TransferRequestBuilder.java (hunk 1 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/TransferRequestBuilder.java:50]: Added Units tests for custom Json and EDC management API in DataSourceParameterizationTest

File: extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/ParameterizationCompatibilityUtils.java (hunk 4 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/ParameterizationCompatibilityUtils.java:44]: ok

**Review by efiege — APPROVED** [review]
LGTM, awesome work!

**Pull request merged**